src/demo/data.js

```diff
@@ -2,51 +2,178 @@ import faker from 'faker/locale/en';
+  enrollment_count: getEnrollmentsForUser(enrollment.user_email).length,
+  course_completion_count: getCompletedEnrollmentsForUser(enrollment.user_email).length,
+  last_activity_date: enrollment.last_activity_date,
+}));
+
+const filterUnenrolledRegisteredLearners = () => enrollments.map(enrollment => ({
+  user_email: enrollment.user_email,
+  user_account_creation_timestamp: faker.date.past(),
+}));
+
+const filterLearnerActivity = (filterOption) => {
+  let filteredLearnerEnrollments = enrollments;
+  const lastActivityDate = new Date();
+  if (filterOption === 'active_past_week') {
+    lastActivityDate.setDate(lastActivityDate.getDate() - 7);
+    filteredLearnerEnrollments = enrollments.filter(enrollment => (
+      enrollment.last_activity_date.getTime() >= lastActivityDate.getTime()
+    ));
+  } else if (filterOption === 'inactive_past_week') {
+    lastActivityDate.setDate(lastActivityDate.getDate() - 7);
+    filteredLearnerEnrollments = enrollments.filter(enrollment => (
+      enrollment.last_activity_date.getTime() <= lastActivityDate.getTime()
+    ));
+  } else if (filterOption === 'inactive_past_month') {
+    lastActivityDate.setDate(lastActivityDate.getDate() - 30);
+    filteredLearnerEnrollments = enrollments.filter(enrollment => (
+      enrollment.last_activity_date.getTime() <= lastActivityDate.getTime()
+    ));
+  }
+  return filteredLearnerEnrollments;
+};
+
+const filterPastWeekCompletions = () => {
+  const today = new Date();
+  const passedDateTime = new Date().setDate(today.getDate() - 7);
+  return completedEnrollments.filter(enrollment => (
+    enrollment.passed_timestamp.getTime() >= passedDateTime &&
+    enrollment.passed_timestamp.getTime() <= today.getTime()
+  ));
+};
+
+const filterCompletedLearnerCourses = () => completedEnrollments.map(enrollment => ({
+  user_email: enrollment.user_email,
+  completed_courses: getEnrollmentsForUser(enrollment.user_email).length,
+}));
+
+const filteredEnrollments = (options) => {
+  let filterEnrollments = enrollments;
+  if (options.learner_activity) {
+    filterEnrollments = filterLearnerActivity(options.learner_activity);
+  }
+  if (options.passed_date) {
+    filterEnrollments = filterPastWeekCompletions();
+  }
+  if (options.has_enrollments && options.extra_fields === 'enrollment_count') {
+    filterEnrollments = filterEnrolledLearners();
+  }
+  return filterEnrollments;
+};
 
 const overview = {
-  // We could derive this data from the enrollments but for now just using static data
   active_learners: {
-    past_month: 111,
-    past_week: 48,
+    past_month: filterPastWeekCompletions().length,
+    past_week: filterLearnerActivity('active_past_week').length,
   },
-  course_completions: 85,
-  enrolled_learners: 203,
-  last_update_date: new Date(),
-  number_of_users: 268,
+  course_completions: filterCompletedLearnerCourses().length,
+  enrolled_learners: filterEnrolledLearners().length,
+  last_update_date: new Date().toString(),
```

Comment:
Thanks for adding the `toString()` here :) Per my previous comment, though, `last_update_date` should be `last_updated_date` as that is the key the API returns.